test/service/test_manifest.py

```diff
@@ -408,7 +412,36 @@ def test_compact_manifest(self):
         self.maxDiff = None
         bundle_fqid = self.bundle_fqid(uuid='f79257a7-dfc6-46d6-ae00-ba4b25313c10',
                                        version='2018-09-14T133314.453337Z')
-        self._index_canned_bundle(bundle_fqid)
+        bundle = self._load_canned_bundle(bundle_fqid)
+        self._index_bundle(bundle)
+
+        # Duplicate one of the files into a minimal mock bundle to test
+        # redundant file contributions from different bundles (for example due
+        # to stitching)
+        manifest = [
+            entry
+            for entry in bundle.manifest
+            if entry['name'] in {
+                'supplementary_file_1.json',
+                'SmartSeq2_RTPCR_protocol.pdf',
+                'links.json',
+                'project_0.json'
+            }
+        ]
+        metadata_files = {
+            document_id: content
+            for document_id, content in bundle.metadata_files.items()
+            if document_id in {
+                'supplementary_file_1.json',
+                'links.json',
+                'project_0.json'
+            }
+        }
+        metadata_files['links.json']['links'].clear()
```

Comment:
This looks wrong to me or deserves an explanation.

Reply:
It also looks like it modifies the mock *and* the original which introduces an additional, brittle constraint that the original is indexed first.